src/lib.rs

```diff
@@ -65,14 +65,111 @@ impl Frame {
+    pub force_integer_mv: bool,      // 0 - Not to force. MV can be in 1/4 or 1/8
+                                     // 1 - force to integer
+                                     // 2 - adaptive
+    pub still_picture: bool,               // Video is a single frame still picture
+    pub reduced_still_picture_hdr: bool,   // Use reduced header for still picture
+    pub monochrome: bool,                  // Monochorme video
+    pub enable_filter_intra: bool,         // enables/disables filterintra
+    pub enable_intra_edge_filter: bool,    // enables/disables corner/edge/upsampling
+    pub enable_interintra_compound: bool,  // enables/disables interintra_compound
+    pub enable_masked_compound: bool,      // enables/disables masked compound
+    pub enable_dual_filter: bool,         // 0 - disable dual interpolation filter
+                                          // 1 - enable vert/horiz filter selection
+    pub enable_order_hint: bool,     // 0 - disable order hint, and related tools
+                                     // jnt_comp, ref_frame_mvs, frame_sign_bias
+                                     // if 0, enable_jnt_comp and
+                                     // enable_ref_frame_mvs must be set zs 0.
+    pub enable_jnt_comp: bool,        // 0 - disable joint compound modes
+                                     // 1 - enable it
+    pub enable_ref_frame_mvs: bool,  // 0 - disable ref frame mvs
+                                     // 1 - enable it
+    pub enable_warped_motion: bool,   // 0 - disable warped motion for sequence
+                                     // 1 - enable it for the sequence
+    pub enable_superres: bool,// 0 - Disable superres for the sequence, and disable
+                              //     transmitting per-frame superres enabled flag.
+                              // 1 - Enable superres for the sequence, and also
+                              //     enable per-frame flag to denote if superres is
+                              //     enabled for that frame.
+    pub enable_cdef: bool,         // To turn on/off CDEF
+    pub enable_restoration: bool,  // To turn on/off loop restoration
+    pub operating_points_cnt_minus_1: usize,
+    pub operating_point_idc: [u16; MAX_NUM_OPERATING_POINTS],
+    pub display_model_info_present_flag: bool,
+    pub decoder_model_info_present_flag: bool,
+    pub level: [[usize; 2]; MAX_NUM_OPERATING_POINTS],	// minor, major
+    pub tier: [usize; MAX_NUM_OPERATING_POINTS],  // seq_tier in the spec. One bit: 0
+                                             // or 1.
 }
 
 impl Sequence {
-    pub fn new() -> Sequence {
+    pub fn new(width: usize, height: usize) -> Sequence {
+        let mut operating_point_idc = [0 as u16; MAX_NUM_OPERATING_POINTS];
+        let mut level = [[1, 2 as usize]; MAX_NUM_OPERATING_POINTS];
+        let mut tier = [0 as usize; MAX_NUM_OPERATING_POINTS];
+
+        for i in 0..MAX_NUM_OPERATING_POINTS {
```

Comment:
Why setting them here again if you already set them just right above?

Reply:
Yeap- right. Some members of Sequence are set when it is constructed and others are set right before it is written to bitstream, i.e. after frame is finished for encoding.
Then, I haven't really decided where to init them, so duplicated. Thanks for catching them.